llvm-external-projects/iree-compiler-api/unittests/tools/compiler_core_test.py

```diff
@@ -0,0 +1,216 @@
+# Lint as: python3
+# Copyright 2020 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import contextlib
+import logging
+import os
+import io
+import tempfile
+import unittest
+
+import iree.compiler.tools
+
+SIMPLE_MUL_ASM = """
+func @simple_mul(%arg0: tensor<4xf32>, %arg1: tensor<4xf32>) -> tensor<4xf32> {
+    %0 = "mhlo.multiply"(%arg0, %arg1) {name = "mul.1"} : (tensor<4xf32>, tensor<4xf32>) -> tensor<4xf32>
+    return %0 : tensor<4xf32>
+}
+"""
```

Comment:
could drop the mhlo requirement from this by using a program like this one: https://github.com/google/iree/blob/main/iree/samples/models/simple_abs.mlir

Reply:
I'll do it later - that is just going to break when they move those ops from std -> arith and one less thing to update then :)

Reply:
Argh, right... we tried to make that sample model as simple as possible, but it will still break x_x